frontend/src/components/explorer/gemBrowser/Reaction.vue

```diff
@@ -188,8 +188,15 @@ export default {
             if (e === 'or' || e === 'and') {
               return e;
             }
-            const prefix = e[0] === '(' ? '(' : '';
-            const suffix = e.slice(-1) === ')' ? ')' : '';
+            let prefix = '';
+            let suffix = '';
+            if (e[0] === '(') {
+              prefix = e[1] === '(' ? '((' : '(';
+            }
+            if (e.slice(-1) === ')') {
+              suffix = e.slice(-2) === '))' ? '))' : ')';
+            }
```

Comment:
I've `grep`-ed the `data-files/integrated-models` and it looks like there are cases of `(((` as well. How about using a regex instead,  like the one used below ``` ^\(+|\)+$/g ``` ?

Reply:
Ah, nice that you caught this and sounds like a good suggestion, thanks!

Reply:
I realize that there was a lot of redundant code, which has been removed now in [this commit](https://github.com/MetabolicAtlas/MetabolicAtlas/pull/625/commits/b3af2639dcbd211b0bded9d467fe089fa1a864f6). It looks like the gene rule is only being used in `Reaction.vue`, so instead of reformatting the gene rule first in `frontend/src/api/reactions.js` and then in `frontend/src/components/explorer/gemBrowser/Reaction.vue`, it is only being formatted in the latter now. The formatting logic has also been rewritten to handle any level of nesting. There is also some legacy code involving `geneRule_wname` that has been removed since it's no longer used.

You should be able to test the reaction `/explore/Fruitfly-GEM/gem-browser/reaction/ATPasel` and see the following:

<img width="1001" alt="image" src="https://user-images.githubusercontent.com/423498/123670929-0649bd80-d83e-11eb-9766-7ef289f3d571.png">